src/Eshopworld.Core/IMessenger.cs

```diff
@@ -25,11 +25,17 @@ public interface IPublishEvents : IDisposable
         /// <summary>
         /// Sends an event onto a topic.
         /// </summary>
-        /// <remarks>If <paramref name="topicName"/> is not provided then the event type is used to determine the topic</remarks>
+        /// <typeparam name="T">The type of the event that we are sending.</typeparam>
+        /// <param name="event">The event that we are sending.</param>
+        Task Publish<T>(T @event) where T : class;
+
+        /// <summary>
+        /// Sends an event onto a topic.
+        /// </summary>
         /// <typeparam name="T">The type of the event that we are sending.</typeparam>
         /// <param name="event">The event that we are sending.</param>
         /// <param name="topicName">The name of the topic we are sending to</param>
-        Task Publish<T>(T @event, string? topicName = null) where T : class;
```

Comment:
```csharp
string topicName = null
```

Reply:
> @macuistin can you simply replace `string?` by `string` on the interfaces? We will then need to update the messaging library as well so it uses the latest from Core.

I wen to use the latest Messenger package and it broke an API, as a package in the dependency chain was compiled against an earlier version.
After a lot of investigation, the problem turned out to be those optional parameters were added to public interfaces.
While this doesn't break at compile-time, it does break contracts at runtime, so the original interface method lookups do not work.
I spoke with @mayankgoyel and @C-M-Ryan, and the two options to fix the interface issue in 4.2.0, was to find all existing uses of the Core package and update them, but that would include a lot of internal code bases. The other option was to roll back the change and add overloads so that the required functionality was available while not breaking the contract for existing code.

I compiled and used this version of the binary locally, so I know that it now does not break dependencies that pick this later version up. So that I could also use the Messenger, I also created a draft PR to fix that:
https://github.com/eShopWorld/messaging/pull/24

CC: @chomick